### PR TITLE
Display budget info even if no time has been tracked to that task.

### LINKED ISSUE #5: Show task budget even if no hours have been logged.
Currently we're grabbing the budgeted hours from the first time entry logged to the task. It'd be an extra XHR, but I think we could use the `task_assignments` API to get the budget separate of any time entries (like we're doing in `set_task.ts`).

### PR DESCRIPTION
**Pull request by jgerigmeyer**
## Cute animal pic
_Because everyone likes pictures of animals._

snow leopard

## Description
_The commit messages say what you did; this should explain why and how._

- Fixes #5 

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally._

- Add a new task with a budget but no time logged, and attach to a Trello card.
- Tested that you can now see the budget on the card.

## Show me
_Provide screenshots/animated gifs/videos if necessary._

Screenshot 2019-11-02 15 08 59

---

## Reviewer tasks:

- [ ] Pulled branch and ran code locally (or viewed on preview deployment, if applicable)
- [ ] Reproduced the issue to review in the browser
- [x] Reviewed code

### CODE REVIEW

File: src/lib/harvest.ts [hunk 1]
```diff
@@ -15,3 +18,20 @@ export const getHarvestJSON = (t: Trello, path: string) =>
         },
       }).then((response) => response.json()),
   );
+
+// @@@ TODO if we ever assign more than 100 tasks to a single project,
+// this will break due to API pagination. So let's not do that, m'kay.
+export const getTaskAssignments = (
+  t: Trello,
+  projectId: number,
+): BluebirdPromise<TaskAssignmentsData> =>
```
Comment (carljm) [inline src/lib/harvest.ts]: We can't use TrelloPromise for this annotation?

File: src/lib/types.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  billable: boolean;
+  created_at: string;
+  updated_at: string;
+  hourly_rate: number | null;
+  budget: number | null;
+  project: { id: number; name: string; code: string };
+  task: { id: number; name: string };
+}
+
+export interface TaskAssignmentsData extends HarvestAPIResponse {
```
Comment (carljm) [inline src/lib/types.ts]: nit: I'd rather call this type `TaskAssignmentsResponse`

File: src/lib/types.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  external_reference: {
+    id: string;
+    group_id: string;
+    permalink: string;
+    service: string;
+    service_icon_url: string;
+  };
+}
+
+export interface TimeEntriesData extends HarvestAPIResponse {
```
Comment (carljm) [inline src/lib/types.ts]: and this one `TimeEntriesResponse`

File: src/lib/types.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  fee: number | null;
+  notes: string;
+  starts_on: string | null;
+  ends_on: string | null;
+  created_at: string;
+  updated_at: string;
+  client: { id: number; name: string; currency: string };
+}
+
+export interface ProjectsData extends HarvestAPIResponse {
```
Comment (carljm) [inline src/lib/types.ts]: And this one `ProjectsResponse`

**Review by carljm — APPROVED** [review]
Don't have time to set up dev right now to test the change functionally, but the code looks great and I trust your functional testing!

**Comment by carljm** [comment]
My reasoning for using the name `*Response` for all the response types is that they inherit `HarvestAPIResponse` type. But if you feel like "Response" is not the right name for them (e.g. because they aren't real fetch `Response` objects, they are just the data from the response body), I can see that argument; in that case for consistency I'd just want to rename `HarvestAPIResponse` to `HarvestAPIResponseData`.

File: src/lib/harvest.ts (hunk 1 above)
Comment (jgerigmeyer) [reply src/lib/harvest.ts]: We've already defined the value TrelloPromise as returning a BluebirdPromise, so I think this is actually more accurate?

**Pull request merged**